testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CLITestCase.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ *
+ */
+package org.jboss.as.test.integration.management.cli;
+
+import org.jboss.as.cli.CommandContext;
+import org.jboss.as.cli.scriptsupport.CLI;
+import org.jboss.as.cli.scriptsupport.CLI.Result;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test various connection states of the CLI class.
+ *
+ * @author Jean-Francois Denise (jdenise@redhat.com)
+ */
+public class CLITestCase {
+
+    @Test
+    public void testConnectStatus() {
+
+        // Offline instance
+        CLI cli = CLI.newInstance();
+
+        // Must fail to connect to remote management disabled server
+        // WFCORE-1471 Uncomment when this bug is fixed
+        //checkFail(() -> cli.connect(TestSuiteEnvironment.getServerAddress(),
+        //        TestSuiteEnvironment.getServerPort(), null, null));
+
+        // start an embedded server
+        executeCommand(cli, "embed-server --std-out=echo");
+
+        if (cli.getCommandContext().getControllerPort() > 0) {
+            Assert.fail("Invalid port for embedded "
+                    + cli.getCommandContext().getControllerPort());
+        }
+
+        // Enable management
+        executeCommand(cli, "reload --admin-only=false");
+
```

Comment:
We are aware of the issue and @jfdenise is going to follow up with a fix. Apparently, instead of the reload command, what gets executed is :reload operation, so the logic for making sure the controller is up in the reload handler is not run.
Sorry about that. And thanks for the tips!